gradle-jdks/src/main/java/com/palantir/gradle/jdks/SynchronizedInterface.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+final class SynchronizedInterface {
```

Comment:
My biggest concern with synchronising everything is causing a deadlock somehow.

Reply:
Ya....most of these method implementations are fairly simple, there are just some which change state of the internal object but do it in a very unsafe way. This is a lot cleaner than the monstrosity I had.

Reply:
Yeah, stepping through what methods are called on a projects to these types it doesn't seem that wild.